src/drivers/angular2.js

```diff
@@ -47,62 +48,74 @@ const equals = (obj1, obj2) => {
 
 export const angular2 : ComponentDriverType<*, Angular2, Angular2Module, *> = {
 
-    register: (tag, propsDef, init, { Component : AngularComponent, NgModule, ElementRef, NgZone }) => {
-        const getProps = (component) => {
-            return replaceObject({ ...component.internalProps, ...component.props }, item => {
-                if (typeof item === 'function') {
-                    return function angular2Wrapped() : void {
-                        // $FlowFixMe
-                        return component.zone.run(() => item.apply(this, arguments));
-                    };
+    register: (tag, propsDef, init, { Component : AngularComponent, NgModule, ElementRef, NgZone, Inject }) => {
+        class ComponentInstance {
+            elementRef : Object;
+            internalProps : Object;
+            parent : Object;
+            props : Object;
+            zone : Object;
+            _props : Object;
+
+            static annotations : $ReadOnlyArray<*>;
+            static parameters : $ReadOnlyArray<*>;
+
+            constructor (elementRef, zone) {
+                this._props = {};
+                this.elementRef = elementRef;
+                this.zone = zone;
+            }
+
+            getProps () : Object {
+                return replaceObject({ ...this.internalProps, ...this.props }, item => {
+                    if (typeof item === 'function') {
+                        const { zone } = this;
+                        return function angular2Wrapped() : void {
+                            // $FlowFixMe
+                            return zone.run(() => item.apply(this, arguments));
+                        };
+                    }
+                    return item;
+                });
+            }
+
+            ngOnInit() {
+                const targetElement = this.elementRef.nativeElement;
```

Comment:
Should we null check this or provide validation if `elementRef` is not passed?

Reply:
I am not sure if I see a scenario in which `elementRef` is not passed.
The element `elementRef` is injected by the Angular lib itself on our constructor.

Are you seeing something that I am not?
If you think that we need that anyway, just let me know and I will be happy to add it.